### ISSUE TITLE
bigip_virtual_server ip_protocol doesn't accept 'any' as valid choice

### ISSUE THREAD

**Issue by c41m4n** [F5Networks/f5-ansible#1272]
##### ISSUE TYPE

 - Bug Report

##### ANSIBLE VERSION

```
$ [2] ansible --version
ansible 2.7.8
  config file = /home/me/ansible-datacenter-prod/ansible.cfg
  configured module search path = [u'/home/me/ansible-datacenter-prod/my_modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

```

##### PYTHON VERSION

` $ [0] python -V
Python 2.7.5
``

```

##### BIGIP VERSION
<!---
Paste verbatim output from “tmsh show sys version” between quotes below
-->
```
 tmsh show sys version

Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.4
  Build       0.0.4
  Edition     Point Release 4
  Date        Wed Jan 16 18:04:36 PST 2019

```

##### CONFIGURATION
<!---
Mention any settings you have changed/added/removed in ansible.cfg
(or using the ANSIBLE_* environment variables).
Mention the version of the f5ansible galaxy role used in testing.
-->

##### OS / ENVIRONMENT
<!---
Mention the OS you are running Ansible from, and the OS you are
managing, or say “N/A” for anything that is not platform-specific.
-->
RHEL7
##### SUMMARY
<!--- Explain the problem briefly -->
Documentation linked here: https://clouddocs.f5.com/products/orchestration/ansible/devel/modules/bigip_virtual_server_module.html#bigip-virtual-server-module

Specifies that I can use 'any' as a valid input for the ip_protocol parameter but when I do so the play errors out.
##### STEPS TO REPRODUCE
<!---
For bugs, show exactly how to reproduce the problem, providing a test playbook using 
latest f5ansible galaxy role. 
This is a requirement for the issue to be accepted and resolved efficiently.
For new features, show how the feature would be used.

-->

<!--- Paste example playbooks or commands between quotes below -->
```
- name: "create-inbound-afm-policy: Create inboundv4 VS"
  delegate_to: localhost
  bigip_virtual_server:
    provider: "{{ f5provider }}"
    state: present
    partition: Common
    name: "SYS-{{ data_class|upper }}-{{ tenant|upper }}-INBOUND-IPv4-FWD_VS"
    description: "Inbound VLAN {{ vlan_id }} traffic"
    source: 0.0.0.0/0
    port: 0
    ip_protocol: any
    destination: "{{ ipv4_network }}"
    enabled_vlans: "{{ outside_vlan_name }}"
    type: forwarding-ip
    firewall_enforced_policy: "{{ data_class|upper }}-{{ tenant|upper }}-INBOUNDv4_AFM_POLICY"
    security_log_profiles: log-network

```

<!--- You can also paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- What did you expect to happen when running the steps above? -->
The default value when not specifying ip_protocol is TCP but I want to match all protocols.
##### ACTUAL RESULTS
<!--- What actually happened? If possible run with high verbosity (-vvvv) -->

<!--- Paste verbatim command output between quotes below -->
```
TASK [f5-create-subnet : create-inbound-afm-policy: Create inboundv4 VS] *****************************************************************************************************************************************************************
task path: /home/XXXXXXXXX/ansible-datacenter-prod/roles/f5-create-subnet/tasks/05-create-inboundv4-afm-policy.yml:18
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: XXXXXXXXX
<localhost> EXEC /bin/sh -c 'echo ~XXXXXXXXX && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.93-167404003515966 `" && echo ansible-tmp-1553293019.93-167404003515966="` echo /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-155329$
019.93-167404003515966 `" ) && sleep 0'
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: XXXXXXXXX
<localhost> EXEC /bin/sh -c 'echo ~XXXXXXXXX && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.94-222796425230939 `" && echo ansible-tmp-1553293019.94-222796425230939="` echo /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-155329$
019.94-222796425230939 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/network/f5/bigip_virtual_server.py
<localhost> PUT /home/XXXXXXXXX/.ansible/tmp/ansible-local-23319EZNjKg/tmpQr5FJL TO /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.94-222796425230939/AnsiballZ_bigip_virtual_server.py
Using module file /usr/lib/python2.7/site-packages/ansible/modules/network/f5/bigip_virtual_server.py
<localhost> PUT /home/XXXXXXXXX/.ansible/tmp/ansible-local-23319EZNjKg/tmpN9mAwN TO /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.93-167404003515966/AnsiballZ_bigip_virtual_server.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.94-222796425230939/ /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.94-222796425230939/AnsiballZ_bigip_virtual_server.py && sleep 0'
<localhost> EXEC /bin/sh -c 'chmod u+x /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.93-167404003515966/ /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.93-167404003515966/AnsiballZ_bigip_virtual_server.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.94-222796425230939/AnsiballZ_bigip_virtual_server.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.93-167404003515966/AnsiballZ_bigip_virtual_server.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/XXXXXXXXX/.ansible/tmp/ansible-tmp-1553293019.93-167404003515966/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_bigip_virtual_server_payload_apME71/__main__.py", line 640, in <module>
    from library.module_utils.network.f5.bigip import F5RestClient

fatal: [lb-test-b06-mgmt.XXXXXXXXX -> localhost]: FAILED! => changed=false
  invocation:
    module_args:
      description: Inbound VLAN 3699 traffic
      destination: 10.92.255.0
      enabled_vlans:
      - DC_NFW_F5-TEST_NPUB
      firewall_enforced_policy: TEST-F5-TEST-INBOUNDv4_AFM_POLICY
      ip_protocol: any
      name: SYS-TEST-F5-TEST-INBOUND-IPv4-FWD_VS
      partition: Common
      port: '0'
      provider:
        auth_provider: null
        password: XXXXXXXXX
        server: lb-test-b06-mgmt.XXXXXXXXX
        server_port: null
        ssh_keyfile: null
        timeout: null
        transport: rest
        user: ansible
        validate_certs: false
      security_log_profiles:
      - XXXXXXXXX-network
      source: 0.0.0.0/0
      state: present
      type: forwarding-ip
  msg: 'value of ip_protocol must be one of: ah, bna, esp, etherip, gre, icmp, ipencap, ipv6, ipv6-auth, ipv6-crypt, ipv6-icmp, isp-ip, mux, ospf, sctp, tcp, udp, udplite, got: any'

```

**Comment by caphrim007** [comment]
I think this is probably already addressed, as per this line here

https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_virtual_server.py#L3552

In the version of Ansible that you have reported (2.7.8) this feature had not yet been added. Given the above code line (and the associated addition to the Ansible core product) you'll be able to resolve this issue by either adopting the "role" method of using the modules in this repository, or, waiting for the release of Ansible 2.8 sometime in May

**Comment by DerrickTJ** [comment]
While @caphrim007 highlighted policies in the link above, it was actually profiles that fixes the issue in 2.8. ipother must be set if you are using a protocol of any on the VIP. Of course you can set it to something else if you've created a different profile. ipother is just the only one that is default on the system.

```    
- name: Add virtual server
      bigip_virtual_server:
        state: present
        name: DerrickTest
        destination: 10.10.10.10
        port: 0
        ip_protocol: any
        description: Test Virtual Server
        profiles:
          - ipother
```
